### PR TITLE
added model.save and model.restore for "tensorflow" resolves #648

### LINKED ISSUE #648: [ENH] Add model.save() to TensorFlow 2.0
I'll work on this issue. Best wishes!
Paul
https://github.com/lululxvi/deepxde/blob/f72ac3e76e3efc25ba63469c14f48ee2a15cfee7/deepxde/model.py#L711

### PR DESCRIPTION
**Pull request by pescap**

### CODE REVIEW

**Comment by pescap** [comment]
Tested with:

```
"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch"""
import deepxde as dde
import matplotlib.pyplot as plt
import numpy as np
# Import tf if using backend tensorflow.compat.v1 or tensorflow
from deepxde.backend import tf
# Import torch if using backend pytorch
# import torch


def pde(x, y):
    dy_xx = dde.grad.hessian(y, x)
    # Use tf.sin for backend tensorflow.compat.v1 or tensorflow
    return -dy_xx - np.pi ** 2 * tf.sin(np.pi * x)
    # Use torch.sin for backend pytorch
    # return -dy_xx - np.pi ** 2 * torch.sin(np.pi * x)


def boundary(x, on_boundary):
    return on_boundary


def func(x):
    return np.sin(np.pi * x)


geom = dde.geometry.Interval(-1, 1)
bc = dde.icbc.DirichletBC(geom, func, boundary)
data = dde.data.PDE(geom, pde, bc, 16, 2, solution=func, num_test=100)

layer_size = [1] + [50] * 3 + [1]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001, metrics=["l2 relative error"])

#losshistory, train_state = model.train(epochs=10000)
# Optional: Save the model during training.
name = "model/model"
checkpointer = dde.callbacks.ModelCheckpoint(name, verbose=1, save_better_only=True)
# Optional: Save the movie of the network solution during training.
# ImageMagick (https://imagemagick.org/) is required to generate the movie.
# movie = dde.callbacks.MovieDumper(
#     "model/movie", [-1], [1], period=100, save_spectrum=True, y_reference=func
# )
losshistory, train_state = model.train(epochs=2000, callbacks=[checkpointer])

dde.saveplot(losshistory, train_state, issave=True, isplot=True)
model.restore(name+'-'+str(train_state.best_step) + ".ckpt", verbose=1)
model.restore(name+"-1000.ckpt", verbose=1)
```

File: deepxde/model.py [hunk 1]
```diff
@@ -684,6 +684,7 @@ def save(self, save_path, protocol="backend", verbose=0):
             save_path (string): Prefix of filenames to save the model file.
             protocol (string): If `protocol` is "backend", save using the backend-specific method.
                 For "tensorflow.compat.v1", use `tf.train.Save <https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/Saver#attributes>`_.
+                For "tensorflow", use tf.keras.models.save_weights <https://www.tensorflow.org/api_docs/python/tf/keras/Model#save_weights>`_.
```
Comment (pescap) [inline deepxde/model.py:687]:
```suggestion
                For "tensorflow", use tf.keras.Model.save_weights <https://www.tensorflow.org/api_docs/python/tf/keras/Model#save_weights>`_.
```

File: deepxde/model.py [hunk 2]
```diff
@@ -684,6 +684,7 @@ def save(self, save_path, protocol="backend", verbose=0):
             save_path (string): Prefix of filenames to save the model file.
             protocol (string): If `protocol` is "backend", save using the backend-specific method.
                 For "tensorflow.compat.v1", use `tf.train.Save <https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/Saver#attributes>`_.
+                For "tensorflow", use tf.keras.Model.save_weights <https://www.tensorflow.org/api_docs/python/tf/keras/Model#save_weights>`_.
```
Comment (lululxvi) [inline deepxde/model.py:687]:
```
For "tensorflow", use `tf.keras.Model.save_weights <https://www.tensorflow.org/api_docs/python/tf/keras/Model#save_weights>`_.
```

Comment (pescap) [inline deepxde/model.py:687]:
```suggestion
                For "tensorflow", use `tf.keras.Model.save_weights <https://www.tensorflow.org/api_docs/python/tf/keras/Model#save_weights>`_.
```

**Pull request merged**